src/server.go

```diff
@@ -76,9 +77,37 @@ func (a *App) QueryHandler(key []byte, w http.ResponseWriter, r *http.Request) {
   }
 }
 
+func (a *App) CheckAuthorization(w http.ResponseWriter, r *http.Request) (string, bool) {
+  var authstring string
+  if a.htpasswdfile != nil {
+    authtoken := r.Header.Get("Authorization")
+    if authtoken != "" {
+      authstringb, _ := base64.StdEncoding.DecodeString(strings.Split(authtoken, "Basic ")[1])
+      authstring = string(authstringb)
+      username := strings.Split(authstring, ":")[0]
+      password := strings.Split(authstring, ":")[1]
+      if !a.htpasswdfile.Match(username, password) {
+        w.WriteHeader(401)
+        return authstring, false
+      }
+      authstring = authstring+"@"
+    } else {
+      w.WriteHeader(401)
+      return authstring, false
+    }
+  }
+  return authstring, true
+}
+
 func (a *App) ServeHTTP(w http.ResponseWriter, r *http.Request) {
   key := []byte(r.URL.Path)
 
+  // check if user is authorized
+  authstring, authorized := a.CheckAuthorization(w, r)
```

Comment:
Can we outsource the auth check to the volume nodes? I don't like having it in two places

Reply:
Ok, let me see if I understand clearly. You are suggesting that the basic auth is only checked at the nginx, right? I.e., the go server accepts every incoming request, checks with a volume whether that authentication is valid or not, and then proceeds with the logic. That way we don't need any other dependency, and the server is still protected